packages/remix-dev/config/flat-routes.ts

```diff
@@ -124,6 +131,18 @@ export function getRouteSegments(routeId: string) {
   let routeSegment = "";
   let rawRouteSegment = "";
   let state: State = "NORMAL";
+  let hasFolder = /[/\\]/.test(routeId);
+
+  /**
+   * @see https://github.com/remix-run/remix/pull/5160#issuecomment-1402157424
+   */
+  if (hasFolder && (routeId.endsWith("/index") || routeId.endsWith("/route"))) {
+    let last = routeId.lastIndexOf(path.posix.sep);
+    if (last >= 0) {
+      routeId = routeId.substring(0, last);
+    }
+  }
+
```

Comment:
check if we have a routeId with a folders, if we do, drop `index` and `route` segments